Updating to pnpm is broken

- Description of the bug
Updating a Vaadin 14.1 project to 14.2 does not work

- Minimal reproducible example
1. Clone https://github.com/vaadin/skeleton-starter-flow/tree/broken_pnpm
2. run `mvn clean jetty:run`

- Expected behavior
The app starts using pnpm and works 

- Actual behavior
Error on console:
```
[main] INFO com.vaadin.flow.server.startup.DevModeInitializer - Starting dev-mode updaters in C:\Users\Pekka\workspaces\flow\skeleton-starter-flow folder.
[main] INFO dev-updater - Visited 92 classes. Took 27 ms.
[main] INFO dev-updater - Added 38 dependencies to 'C:\Users\Pekka\workspaces\flow\skeleton-starter-flow\target\frontend\package.json'
[main] INFO dev-updater - Removed 1 dependencies
[main] INFO dev-updater - Updated npm C:\Users\Pekka\workspaces\flow\skeleton-starter-flow\package.json.
[main] INFO dev-updater - Running `pnpm install` ...
npm WARN saveError ENOENT: no such file or directory, open 'C:\Users\Pekka\workspaces\flow\skeleton-starter-flow\package.json'
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN enoent ENOENT: no such file or directory, open 'C:\Users\Pekka\workspaces\flow\skeleton-starter-flow\package.json'
npm WARN skeleton-starter-flow No description
npm WARN skeleton-starter-flow No repository field.
npm WARN skeleton-starter-flow No README data
npm WARN skeleton-starter-flow No license field.

+ pnpm@4.5.0
added 1 package from 1 contributor and audited 1 package in 2.678s
found 0 vulnerabilities

Packages: +1
+
Resolving: total 1, reused 1, downloaded 0, done

dependencies:
+ pnpm 4.5.0
[main] ERROR dev-updater - >>> Dependency ERROR. Check that all required dependencies are deployed in pnpm repositories.
[WARNING]
javax.servlet.ServletException: com.vaadin.flow.server.ExecutionFailedException: Pnpm install has exited with non zero status. Some dependencies are not installed. Check pnpm command output
    at com.vaadin.flow.server.startup.DevModeInitializer.initDevModeHandler (DevModeInitializer.java:295)
    at com.vaadin.flow.server.startup.DevModeInitializer.onStartup (DevModeInitializer.java:185)
    at org.eclipse.jetty.plus.annotation.ContainerInitializer.callStartup (ContainerInitializer.java:140)
    at org.eclipse.jetty.annotations.ServletContainerInitializersStarter.doStart (ServletContainerInitializersStarter.java:64)
...
Caused by: com.vaadin.flow.server.ExecutionFailedException: Pnpm install has exited with non zero status. Some dependencies are not installed. Check pnpm command output
    at com.vaadin.flow.server.frontend.TaskRunNpmInstall.runNpmInstall (TaskRunNpmInstall.java:110)
    at com.vaadin.flow.server.frontend.TaskRunNpmInstall.execute (TaskRunNpmInstall.java:56)
    at com.vaadin.flow.server.frontend.NodeTasks.execute (NodeTasks.java:432)
    at com.vaadin.flow.server.startup.DevModeInitializer.initDevModeHandler (DevModeInitializer.java:284)
    at com.vaadin.flow.server.startup.DevModeInitializer.onStartup (DevModeInitializer.java:185)
    at org.eclipse.jetty.plus.annotation.ContainerInitializer.callStartup (ContainerInitializer.java:140)
    at org.eclipse.jetty.annotations.ServletContainerInitializersStarter.doStart (ServletContainerInitializersStarter.java:64)
```

- Versions:
    - Vaadin / Flow version: 14.2-SNAPSHOT
    - Java version:  1.8.0_212, vendor: AdoptOpenJDK
    - OS version: Windows 10

--- a/flow-server/src/main/java/com/vaadin/flow/server/frontend/FrontendUtils.java
+++ b/flow-server/src/main/java/com/vaadin/flow/server/frontend/FrontendUtils.java
@@ -50,6 +50,7 @@ import com.vaadin.flow.server.frontend.FallbackChunk.CssImportData;
 
 import elemental.json.JsonArray;
 import elemental.json.JsonObject;
+
 import static com.vaadin.flow.server.Constants.SERVLET_PARAMETER_STATISTICS_JSON;
 import static com.vaadin.flow.server.Constants.STATISTICS_JSON_DEFAULT;
 import static com.vaadin.flow.server.Constants.VAADIN_SERVLET_RESOURCES;
@@ -793,20 +794,6 @@ public class FrontendUtils {
             // install pnpm locally using npm
             installPnpm(baseDir, getNpmExecutable(baseDir));
 
-            /*
-             * pnpm is not able to manage itself. To be able to deal with it
-             * properly let's install it now via itself
-             */
-            installPnpm(baseDir, getPnpmExecutable(baseDir, true));
-
-            /*
-             * pnpm installed by npm should have been moved to another location
-             * which is out of regular "node_modules" package root. We don't
-             * need anymore pnpm installed inside "node_modules" and it's even
-             * dangerous to have it there since it's not able to manage itself
-             * properly. Let's remove it from "node_modules".
-             */
-            new File(baseDir, PNMP_INSTALLED_BY_NPM_FOLDER).delete();
             // remove package-lock.json which contains pnpm as a dependency.
             new File(baseDir, "package-lock.json").delete();
 
@@ -838,7 +825,7 @@ public class FrontendUtils {
         List<String> command = new ArrayList<>();
         command.addAll(installCommand);
         command.add("install");
-        command.add("pnpm@4.3.3");
+        command.add("pnpm@4.5.0");
 
         ProcessBuilder builder = createProcessBuilder(command);
         builder.environment().put("ADBLOCK", "1");
@@ -1057,6 +1044,12 @@ public class FrontendUtils {
     }
 
     private static Optional<File> getLocalPnpmScript(String baseDir) {
+        File npmInstalled = new File(baseDir, PNMP_INSTALLED_BY_NPM);
+        if (npmInstalled.canRead()) {
+            return Optional.of(npmInstalled);
+        }
+
+        // For version 4.3.3 check ".ignored" folders
         File movedPnpmScript = new File(baseDir,
                 "node_modules/.ignored_pnpm/bin/pnpm.js");
         if (movedPnpmScript.canRead()) {
@@ -1067,11 +1060,6 @@ public class FrontendUtils {
                 "node_modules/.ignored/pnpm/bin/pnpm.js");
         if (movedPnpmScript.canRead()) {
             return Optional.of(movedPnpmScript);
-        }
-
-        File npmInstalled = new File(baseDir, PNMP_INSTALLED_BY_NPM);
-        if (npmInstalled.canRead()) {
-            return Optional.of(npmInstalled);
         }
         return Optional.empty();
     }

--- a/flow-server/src/test/java/com/vaadin/flow/server/frontend/FrontendUtilsTest.java
+++ b/flow-server/src/test/java/com/vaadin/flow/server/frontend/FrontendUtilsTest.java
@@ -303,13 +303,6 @@ public class FrontendUtilsTest {
             Assert.assertFalse(
                     FrontendUtils.getPnpmExecutable(baseDir, false).isEmpty());
 
-            // PNPM is moved to ignored location
-            Assert.assertTrue(
-                    new File(baseDir, "node_modules/.ignored/pnpm/bin/pnpm.js")
-                            .exists()
-                            || new File(baseDir,
-                                    "node_modules/.ignored_pnpm/bin/pnpm.js")
-                                            .exists());
             // locally installed pnpm (via npm/pnpm) is removed
             Assert.assertFalse(new File("node_modules/pnpm").exists());
 

